include/eve/detail/function/simd/riscv/to_logical.hpp

```diff
@@ -0,0 +1,26 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/as_register.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/detail/category.hpp>
+#include <eve/forward.hpp>
+#include <eve/traits/as_logical.hpp>
+
+namespace eve::detail
+{
+template<typename T, typename N>
+EVE_FORCEINLINE auto
+to_logical(wide<T, N> v) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  return v != static_cast<T>(0);
+}
```

Comment:
I'm surprised that you need this, but it's a to_logical problem, not your problem.